static/js/hash_util.js

```diff
@@ -272,3 +273,11 @@ export function is_spectator_compatible(hash) {
 export function current_hash_as_next() {
     return `next=/${encodeURIComponent(window.location.hash)}`;
 }
+
+export function build_login_link() {
+    let login_link = "/login/?" + current_hash_as_next();
+    if (page_params.development_environment) {
+        login_link = "/devlogin/?" + current_hash_as_next();
+    }
+    return login_link;
+}
```

Comment:
This works, though it'd probably be nicer if we arranged it so that hovering the login button displayed the URL it'd go to. Not sure it's worth changing.

Reply:
You can see the link at the left-bottom corner of the screen for most of these buttons except the right sidebar, which is handled separately as it is not re-rendered with narrow change.